Added mimir-continuous-test doc and CHANGELOG entry (#1709)
grafana/mimir: docs/sources/operators-guide/tools/mimir-continuous-test.md

---
title: "Grafana mimir-continuous-test"
menuTitle: "Mimir-continuous-test"
description: "Use mimir-continuous-test to continuously run smoke tests on live Grafana Mimir clusters."
weight: 30
---

# Grafana mimir-continuous-test

As a developer, you can use the standalone mimir-continuous-test tool to continuously run smoke tests on live Grafana Mimir clusters.
This tool aims to help identifying a class of bugs that could be difficult to spot during development with unit and integration tests.

## Download mimir-continuous-test

- Using Docker:

```bash
docker pull "grafana/mimir-continuous-test:latest"
```

- Using a local binary:

> Download the appropriate [release asset](https://github.com/grafana/mimir/releases/latest) for your operating system and architecture and make it executable.

Author: ```suggestion
Download the appropriate [mimir-continuous-test binary](https://github.com/grafana/mimir/releases/latest) for your operating system and architecture, and make it executable.
```

---
title: "Grafana mimir-continuous-test"
menuTitle: "Mimir-continuous-test"
description: "Use mimir-continuous-test to continuously run smoke tests on live Grafana Mimir clusters."
weight: 30
---

# Grafana mimir-continuous-test

> Mimir-continuous-test is a standalone tool that you can use to continuously run smoke tests on live Grafana Mimir clusters.
> This tool targets Mimir developers.

Reviewer: ```suggestion
As a developer, you can use the standalone mimir-continuous-test tool to continuously run smoke tests on live Grafana Mimir clusters.
```

> The target of this tool is for Mimir developers.

Reviewer 2: [Nit] I just think this reads better:
```suggestion
This tool targets Mimir developers.
```

> This tool aims to help identifying a class of bugs that could be difficult to spot during development with unit and integration tests.

Reviewer: ```suggestion
Via unit and integration tests, this tool identifies a class of bugs that could be difficult to spot during development.
```

Author: Suggestion here is not technically accurate. During development we already use unit and integration tests. This tool helps finding bugs that are difficult to spot during development. I'm going to keep your edit and remove the mention to unit and integration tests.

```suggestion
This tool identifies a class of bugs that could be difficult to spot during development.
```

## Download mimir-continuous-test

- Using Docker:

```bash
docker pull "grafana/mimir-continuous-test:latest"
```

- Using a local binary:

> Download the appropriate [release asset](https://github.com/grafana/mimir/releases/latest) for your operating system and architecture and make it executable.

Reviewer: ```suggestion
Download the appropriate [Grafana Mimir release](https://github.com/grafana/mimir/releases/latest) for your operating system and architecture, and make it executable.
```

For Linux with the AMD64 architecture, execute the following command:

```bash
curl -Lo mimir-continuous-test https://github.com/grafana/mimir/releases/latest/download/mimir-continuous-test-linux-amd64
chmod +x mimir-continuous-test
```

## Configure mimir-continuous-test

> Mimir-continuous-test requires the endpoints of the backend Grafana Mimir clusters and the tenant ID to use for writing and querying testing metrics:

Reviewer 2: [Nit] I think the following simplification reads better:
```suggestion
Mimir-continuous-test requires the endpoints of the backend Grafana Mimir clusters and the tenant ID for writing and querying testing metrics:
```

> - Set `-tests.write-endpoint` to the base endpoint on the write path. The URL should have no trailing slash. The specific API path is appended by the tool to the URL, for example `/api/v1/push` for the remote write API.

Reviewer: ```suggestion
- Set `-tests.write-endpoint` to the base endpoint on the write path. Remove any trailing slash from the URL. The tool appends the specific API path to the URL, for example `/api/v1/push` for the remote-write API.
```

> - Set `-tests.read-endpoint` to the base endpoint on the read path. The URL should have no trailing slash. The specific API path is appended by the tool to the URL, for example `/api/v1/query_range` for range query API.

Reviewer: ```suggestion
- Set `-tests.read-endpoint` to the base endpoint on the read path. Remove any trailing slash from the URL. The tool appends the specific API path to the URL, for example `/api/v1/query_range` for the range-query API.
```

- Set `-tests.tenant-id` to the tenant ID to use to write and read metrics in tests.

> **Note:** You can run `mimir-continuous-test -help` to list all available configuration options.

## How it works

> Mimir-continuous-test periodically runs a suite of tests writing data to Mimir, querying it back, and checking if the query results match the expected ones.

Reviewer: ```suggestion
Mimir-continuous-test periodically runs a suite of tests, writes data to Mimir, queries that data back, and checks if the query results match what is expected.
```

> The tool exposes metrics that can be used to alert on test failures.
> Details about the failed tests are logged.

Reviewer: ```suggestion
The tool exposes metrics that you can use to alert on test failures, and the tool logs the details about the failed tests.
```
Unsure whether the tool logs the details or is something else does that because passive voice makes this ambiguous. Please vet this for technical accuracy.

Author: It's the tool logging details, so your edit is 👍

### Exported metrics

> Mimir-continuous-test exposes the following Prometheus metrics at the `/metrics` endpoint listening on the port configured via the flag `-server.metrics-port`:

Reviewer: ```suggestion
Mimir-continuous-test exposes the following Prometheus metrics at the `/metrics` endpoint listening on the port that you configured via the flag `-server.metrics-port`:
```

{test="<name>"}

# HELP mimir_continuous_test_writes_failed_total Total number of failed write requests.
# TYPE mimir_continuous_test_writes_failed_total counter
mimir_continuous_test_writes_failed_total{test="<name>",status_code="<code>"}

# HELP mimir_continuous_test_queries_total Total number of attempted query requests.
# TYPE mimir_continuous_test_queries_total counter
mimir_continuous_test_queries_total{test="<name>"}

# HELP mimir_continuous_test_queries_failed_total Total number of failed query requests.
# TYPE mimir_continuous_test_queries_failed_total counter
mimir_continuous_test_queries_failed_total{test="<name>"}

# HELP mimir_continuous_test_query_result_checks_total Total number of query results checked for correctness.
# TYPE mimir_continuous_test_query_result_checks_total counter
mimir_continuous_test_query_result_checks_total{test="<name>"}

# HELP mimir_continuous_test_query_result_checks_failed_total Total number of query results failed when checking for correctness.
# TYPE mimir_continuous_test_query_result_checks_failed_total counter
mimir_continuous_test_query_result_checks_failed_total{test="<name>"}
```

### Alerts

> The released [Mimir alerts]({{< relref "../visualizing-metrics/installing-dashboards-and-alerts.md" >}}) include checks on failures tracked by mimir-continuous-test.

Reviewer: ```suggestion
[Grafana Mimir alerts]({{< relref "../visualizing-metrics/installing-dashboards-and-alerts.md" >}}) include checks on failures that mimir-continuous-test tracks.
```
Is it important to say that the alerts are released?

Author: I don't think so, you edit looks good to me

> We recommend using the provided alerts when running mimir-continuous-test.

Reviewer: ```suggestion
When running mimir-continuous-test, use the provided alerts.
```

> The released [Mimir alerts]({{< relref "../visualizing-metrics/installing-dashboards-and-alerts.md" >}}) include checks on failures tracked by mimir-continuous-test.
> We recommend to use the provided alerts when running mimir-continuous-test.

Reviewer 2: [Nit] Language improvement:
```suggestion
The released [Mimir alerts]({{< relref "../visualizing-metrics/installing-dashboards-and-alerts.md" >}}) include checks on failures tracked by mimir-continuous-test.
We recommend using the provided alerts when running mimir-continuous-test.
```
